src/pages/index.vue

```diff
@@ -1,81 +1,114 @@
-<template>
-  <HelloWorld msg="Vite Vue TS" />
+<script setup lang="ts">
+import Button from 'primevue/button'
+import Dropdown from 'primevue/dropdown'
+import InputText from 'primevue/inputtext'
+import { useRouter } from 'vue-router'
+import 'primeicons/primeicons.css'
+import {
+  searchParam,
+  pageSize,
+  pageSizes,
+  displayData,
+  displayedPokemon,
+  currentPage,
+  pokeFavorites,
+  favorites
+} from './variables'
+import { searchButtonHandler, getData } from '@/helpers/homepageHelpers'
+import { saveFavorites } from '@/helpers/storeFavorite'
 
-  <WelcomeItem>
-    <template #icon>
-      <DocumentationIcon />
-    </template>
-    <template #heading>Documentation</template>
+const router = useRouter()
 
-    Vue's
-    <a href="https://vuejs.org/" target="_blank" rel="noopener">official documentation</a>
-    provides you with all information you need to get started.
-  </WelcomeItem>
+const updateDisplayed = async () => {
+  displayData.value = await getData(currentPageSize.value, currentPage.value)
+  displayedPokemon.value = displayData.value.results.map((pokemon) => {
+    return {
+      name: pokemon.name,
+      id: +pokemon.url.split('pokemon/')[1].slice(0, -1),
+      url: `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${pokemon.url.split('pokemon/')[1].slice(0, -1)}.png`
+    }
+  })
+}
 
-  <WelcomeItem>
-    <template #icon>
-      <ToolingIcon />
-    </template>
-    <template #heading>Tooling</template>
+const currentPageSize = computed(() => {
+  const num: number = +pageSize.value
+  // console.log(num)
+  if (isNaN(num)) {
+    return 100000 // This is bad, this should never be a hard coded value!
+  } else {
+    return num
+  }
+})
 
-    This project is served and bundled with
-    <a href="https://vitejs.dev/guide/features.html" target="_blank" rel="noopener">Vite</a>. The
-    recommended IDE setup is
-    <a href="https://code.visualstudio.com/" target="_blank" rel="noopener">VSCode</a> +
-    <a href="https://github.com/johnsoncodehk/volar" target="_blank" rel="noopener">Volar</a>. If
-    you need to test your components and web pages, check out
-    <a href="https://www.cypress.io/" target="_blank" rel="noopener">Cypress</a> and
-    <a href="https://on.cypress.io/component" target="_blank" rel="noopener"
-      >Cypress Component Testing</a
-    >.
+const movePage = (direction: 1 | -1) => {
+  currentPage.value += direction
+  updateDisplayed()
+}
```

Comment:
Splitting this into separate `previousPage` and `nextPage` functions might be a little nicer, though I can see why you would approach it like this